### PR TITLE
[Storage ] `az storage logging`: Fix issue #11969 to refine error message

### LINKED ISSUE #11969: Very ugly error message when using az account logging update command on a Storage Account of kind BlobStorage or sku.tier Premium
If you try to exeute az account logging update with a Premium storage account or with a BlobStorage storage account, you will get an error similar to the following:

az>> az storage logging show --account-name xxxxxxxxxxx --subscription xx-xxx-xxx
The command failed with an unexpected error. Here is the traceback:

HTTPSConnectionPool(host='xxxxxxxxxx.queue.core.windows.net', port=443): Max retries exceeded with url: /?restype=service&comp=properties (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00DFC830>: Failed to establish a new connection: [Errno 11002] getaddrinfo failed',))
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connection.py", line 159, in _new_conn
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\util\connection.py", line 57, in create_connection
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 11002] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connectionpool.py", line 600, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connectionpool.py", line 343, in _make_request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connectionpool.py", line 839, in _validate_conn
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connection.py", line 301, in connect
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connection.py", line 168, in _new_conn
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x00DFC830>: Failed to establish a new connection: [Errno 11002] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\requests\requests\adapters.py", line 449, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\connectionpool.py", line 638, in urlopen
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-hb486cm7\urllib3\urllib3\util\retry.py", line 399, in increment
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='xxxxxxxxxxx.queue.core.windows.net', port=443): Max retries exceeded with url: /?restype=service&comp=properties (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00DFC830>: Failed to establish a new connection: [Errno 11002] getaddrinfo failed',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-multiapi-storage\azure\multiapi\storage\v2018_11_09\common\storageclient.py", line 321, in _perform_request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-multiapi-storage\azure\multiapi\storage\v2018_11_09\common\_http\httpclient.py", line 92, in perform_request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\requests\requests\sessions.py", line 533, in request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\requests\requests\sessions.py", line 646, in send
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\requests\requests\adapters.py", line 516, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='xxxxxxxxx.queue.core.windows.net', port=443): Max retries exceeded with url: /?restype=service&comp=properties (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00DFC830>: Failed to establish a new connection: [Errno 11002] getaddrinfo failed',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\bhulse\.azure\cliextensions\interactive\azext_interactive\azclishell\app.py", line 662, in cli_execute
    result = invocation.execute(args)
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\commands\__init__.py", line 603, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\commands\__init__.py", line 661, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\commands\__init__.py", line 652, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\commands\arm.py", line 750, in show_exception_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\commands\__init__.py", line 631, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\commands\__init__.py", line 306, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli-core\azure\cli\core\__init__.py", line 485, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli\azure\cli\command_modules\storage\operations\logging.py", line 15, in get_logging
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-cli\azure\cli\command_modules\storage\services_wrapper.py", line 29, in get_logging
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-multiapi-storage\azure\multiapi\storage\v2018_11_09\queue\queueservice.py", line 372, in get_queue_service_properties
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-multiapi-storage\azure\multiapi\storage\v2018_11_09\common\storageclient.py", line 430, in _perform_request
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-umfcm_n2\azure-multiapi-storage\azure\multiapi\storage\v2018_11_09\common\storageclient.py", line 361, in _perform_request
azure.common.AzureException: HTTPSConnectionPool(host='xxxxxxxxxxx.queue.core.windows.net', port=443): Max retries exceeded with url: /?restype=service&comp=properties (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00DFC830>: Failed to establish a new connection: [Errno 11002] getaddrinfo failed',))

**I would have expected a simple error message along the lines of:**
**Premium Storage accounts do not support logging**
**BlobStorage storage accounts do not support logging.**
I though I was having a problem with api quotas.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c237378a-a708-8e2f-3c8c-0bc8b623d21c
* Version Independent ID: c4e27b67-9605-60f5-899d-8ffd727079b2
* Content: [az storage logging](https://docs.microsoft.com/en-us/cli/azure/storage/logging?view=azure-cli-latest)
* Content Source: [src/azure-cli/azure/cli/command_modules/storage/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/storage/_help.py)
* Service: **storage**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

Fix #11969 
When storage account is BlobStorage or PremiumStorage, we cannot logging for some storage services and the RetryError will be thrown as shown in the issue, which is not  user friendly.

**Testing Guide**  

1. prepare a blob storage account or premium storage account
2. `az storage logging show`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S170

File: src/azure-cli/azure/cli/command_modules/storage/operations/logging.py [hunk 1]
```diff
@@ -15,7 +15,19 @@ def set_logging(client, log, retention, timeout=None, version=None):
...
+        except (KeyError, AzureException):
+            import sys
+            from azure.cli.core.azlogging import CommandLoggerContext
+            from knack.log import get_logger
+
+            logger = get_logger(__name__)
+            with CommandLoggerContext(logger):
+                logger.error("Your storage account doesn't support logging for %s service. Please change value for "
+                             "--services in your commands.", s.name)
+                sys.exit(1)
```
Comment (qianwens) [inline src/azure-cli/azure/cli/command_modules/storage/operations/logging.py:32]: throw CLIError?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/operations/logging.py]: done

File: src/azure-cli/azure/cli/command_modules/storage/operations/logging.py [hunk 2]
```diff
@@ -15,7 +15,14 @@ def set_logging(client, log, retention, timeout=None, version=None):
...
 
 def get_logging(client, timeout=None):
+    from azure.common import AzureException
+    from knack.util import CLIError
     results = {}
     for s in client:
-        results[s.name] = s.get_logging(timeout)
+        try:
+            results[s.name] = s.get_logging(timeout)
+        except (KeyError, AzureException):
```
Comment (qianwens) [inline src/azure-cli/azure/cli/command_modules/storage/operations/logging.py:24]: what is the error code for AzureException? What if it is other error like 500

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/operations/logging.py:24]: Good catch. For service error like 500, I think it should be CloudError which is from ClientException not AzureException. 
Here what I want to catch is ConnectionError, but Storage SDK wraps it to AzureException without error code.

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/operations/logging.py]: added condition for ex.args type,  which should be MaxRetryError

**Pull request merged**